macos/FluentUI/Separator.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AppKit
+
+@objc(MSFSeparatorStyle)
+public enum SeparatorStyle: Int {
+    case `default`
+    case shadow
+
+    fileprivate var color: NSColor {
+		switch self {
+		case .default:
+			if #available(OSX 10.14, *) {
+				return NSColor.separatorColor
+
+			} else {
+				return NSColor.systemGray
+			}
+		case .shadow:
+			return NSColor.gridColor
+        }
+    }
+}
+
+@objc(MSFSeparatorOrientation)
+public enum SeparatorOrientation: Int {
+    case horizontal
+    case vertical
+}
+
+
+@objc(MSFSeparator)
+open class Separator: NSView {
+	private var orientation: SeparatorOrientation = .horizontal
+	
+	private var style: SeparatorStyle = .default
+
+	@objc public override init(frame frameRect: NSRect) {
+		super.init(frame: frameRect)
+		initialize()
+	}
+	
+	@objc public init(orientation: SeparatorOrientation = .horizontal) {
```

Comment:
>init [](start = 14, length = 4)

this initializer can be achieved with the one below in swift since we have a default value. Is the reason to have it here so that it's available to ObjC? If so, why not make one for the other case (init with just a style)?

I wonder how Swift would decide which initializer to use, since the sintax could point to any of them

Reply:
Good point. I removed the default value, and we only have one initializer now

Reply:
Side note: I wrote the initializers as such because I copied it from the iOS initializers:
```
    @objc public override init(frame: CGRect) {
        super.init(frame: frame)
        initialize(style: .default, orientation: .horizontal)
    }

    @objc public init(style: SeparatorStyle = .default, orientation: SeparatorOrientation = .horizontal) {
        super.init(frame: .zero)
        initialize(style: style, orientation: orientation)
    }

    public required init?(coder aDecoder: NSCoder) {
        preconditionFailure("init(coder:) has not been implemented")
    }
```

Might be worth going back and making sure we have matching initializers later on